setup.py

```diff
@@ -13,92 +13,126 @@
+          read('src', 'zc', 'catalog', 'extentcatalog.rst')
           + '\n\n' +
-          read('src', 'zc', 'catalog', 'stemmer.txt')
+          read('src', 'zc', 'catalog', 'stemmer.rst')
           + '\n\n' +
-          read('src', 'zc', 'catalog', 'legacy.txt')
+          read('src', 'zc', 'catalog', 'legacy.rst')
           + '\n\n' +
-          read('src', 'zc', 'catalog', 'globber.txt')
+          read('src', 'zc', 'catalog', 'globber.rst')
           + '\n\n' +
-          read('src', 'zc', 'catalog', 'callablewrapper.txt')
+          read('src', 'zc', 'catalog', 'callablewrapper.rst')
           + '\n\n' +
-          read('src', 'zc', 'catalog', 'browser', 'README.txt')
+          read('src', 'zc', 'catalog', 'browser', 'README.rst')
           ),
-      keywords = "zope3 i18n date time duration catalog index",
-      classifiers = [
+      keywords="zope3 i18n date time duration catalog index",
+      classifiers=[
           'Development Status :: 5 - Production/Stable',
           'Environment :: Web Environment',
           'Intended Audience :: Developers',
           'License :: OSI Approved :: Zope Public License',
           'Programming Language :: Python',
+          'Programming Language :: Python :: 2',
+          'Programming Language :: Python :: 2.7',
+          'Programming Language :: Python :: 3',
+          'Programming Language :: Python :: 3.4',
+          'Programming Language :: Python :: 3.5',
+          'Programming Language :: Python :: 3.6',
+          'Programming Language :: Python :: Implementation :: CPython',
+          'Programming Language :: Python :: Implementation :: PyPy',
           'Natural Language :: English',
           'Operating System :: OS Independent',
           'Topic :: Internet :: WWW/HTTP',
-          'Framework :: Zope3'],
-      url='http://pypi.python.org/pypi/zc.catalog',
+          'Framework :: Zope3',
+      ],
+      url='http://github.com/zopefoundation/zc.catalog',
       license='ZPL 2.1',
       packages=find_packages('src'),
-      package_dir = {'': 'src'},
+      package_dir={'': 'src'},
       namespace_packages=['zc'],
-      extras_require=dict(
-          test=[
-              'zope.keyreference',
-              'zope.testing',
-              ],
-          browser=[
+      extras_require={
+          'test': tests_require,
+          'browser': [
               'zope.app.form',
               'zope.browsermenu',
-              ],
-          test_browser=[
-              'zope.login',
-              'zope.password',
-              'zope.securitypolicy',
-              'zope.testbrowser',
-              'zope.app.appsetup',
-              'zope.app.catalog',
-              'zope.app.testing',
-              'zope.app.zcmlfiles',
-              ]),
+          ],
+          'stemmer:platform_python_implementation=="CPython" and python_version=="2.7"': [
+              'zopyx.txng3.ext',
+          ],
+          'stemmer:python_version >= "3.3"': [
+
+          ],
```

Comment:
You're right, it's there to avoid a warning, `zc.catalog 2.0.0.dev0 does not provide the extra 'stemmer'`

It turns out the stemmer [does work successfully on PyPy](https://github.com/zopyx/zopyx.txng3.ext/pull/6) so we can relax that restriction. It's probably not fast, but that's better than not having the functionality.

I used ">= '3.3'" here because, in spite not claiming support in the classifiers or running tests on CI, this package does actually install and work on 3.3. People might attempt to install it there. 3.2 and below is no longer supported by pip.